packages/client/src/model/MainShip.ts

```diff
@@ -0,0 +1,160 @@
+import { Gif } from '../utils/gif'
+import { SHIP_ENTITY } from '../utils/consts'
+import { rotateObject } from '../utils/game'
+
+export default class MainShip {
+  private _rightPressed = false
+  private _leftPressed = false
+  private _upPressed = false
+  private _downPressed = false
+  private _rotation = SHIP_ENTITY.rotation
+  private readonly _canvasWidth: number
+  private readonly _canvasHeight: number
+  private readonly _velocity: number
+  private readonly _width: number
+  private readonly _height: number
+  private _x: number
+  private _y: number
+  private _ship: Gif
+  private _shipLeft: Gif
+  private _shipRight: Gif
+
+  constructor(canvasWidth: number, canvasHeight: number) {
+    this._canvasWidth = canvasWidth
+    this._canvasHeight = canvasHeight
+    this._velocity = SHIP_ENTITY.velocity
+
+    this._x = this._canvasWidth / 2
+    this._y = this._canvasHeight - SHIP_ENTITY.bottomIdent
+    this._width = SHIP_ENTITY.width
+    this._height = SHIP_ENTITY.height
+    this._ship = new Gif({ url: SHIP_ENTITY.src })
+    this._shipLeft = new Gif({ url: SHIP_ENTITY.srcLeft })
+    this._shipRight = new Gif({ url: SHIP_ENTITY.srcRight })
+
+    document.addEventListener('keydown', this._onKeydown)
+    document.addEventListener('keyup', this._onKeyup)
+  }
+
+  public draw(ctx: CanvasRenderingContext2D) {
+    this._move()
+    this._checkWalls()
+
+    if (this._rightPressed) {
+      rotateObject(
+        ctx,
+        this._rotation,
+        this._x + this._width / 2,
+        this._y + this._height / 2
+      )
+      ctx.drawImage(
+        this._shipRight.image,
+        this._x,
+        this._y,
+        this._width,
+        this._height
+      )
+    } else if (this._leftPressed) {
+      rotateObject(
+        ctx,
+        -this._rotation,
+        this._x + this._width / 2,
+        this._y + this._height / 2
+      )
+      ctx.drawImage(
+        this._shipLeft.image,
+        this._x,
+        this._y,
+        this._width,
+        this._height
+      )
+    } else {
+      ctx.drawImage(
+        this._ship.image,
+        this._x,
+        this._y,
+        this._width,
+        this._height
+      )
+    }
```

Comment:
Я бы написал так:

```suggestion
   const {_x, _y, _width, _height} = this;

    const drawImageParams = [_x, _y, _width, _height];
    const rotateParams = [_x + _width / 2, _y + _height / 2];

    if (this._rightPressed) {
      rotateObject(
        ctx,
        this._rotation,
        ...rotateParams,
      )

      ctx.drawImage(
        this._shipRight.image,
        ...drawImageParams,
      )
    } else if (this._leftPressed) {
      rotateObject(
        ctx,
        -this._rotation,
        ...rotateParams,
      )

      ctx.drawImage(
        this._shipLeft.image,
        ...drawImageParams,
      )
    } else {
      ctx.drawImage(
        this._ship.image,
        ...drawImageParams,
      )
    }
```
Намного быстрее понимаешь что конкретно в аргументах меняется в каждом ифаке, а что везде одинаково. В целом, это повышение читабельности всё.

Reply:
done